### PR TITLE
macos build: bazel 0.7.0 support and more regular CI output

### PR DESCRIPTION
**Pull request by zuercher**
Support for bazel 0.7.0, but continues to work with bazel 0.6.0. The link option is unfortunate, but I think this must be some change to flags within bazel 0.7.0. As noted in the build file, gperftools has already removed the feature that causes this and the linker flag can be removed after their next release. (See discussion in https://github.com/gperftools/gperftools/issues/901)

Also modifies mac CI build to be more explicit about options (instead of trying to use .bazelrc).

### CODE REVIEW

**Comment by mattklein123** [comment]
LGTM. @jmillikin-stripe or @htuch can you give a quick sanity check?

File: bazel/envoy_build_system.bzl [hunk 1]
```diff
@@ -38,7 +38,11 @@ def envoy_linkopts():
         # TODO(zuercher): build id could be supported via "-sectcreate __TEXT __build_id <file>"
         # The file could should contain the current git SHA (or enough placeholder data to allow
         # it to be rewritten by tools/git_sha_rewriter.py).
-        "@bazel_tools//tools/osx:darwin": [],
+        "@bazel_tools//tools/osx:darwin": [
+            # TODO(zuercher): should be able to remove this after the next gperftools release after
```
Comment (jmillikin-stripe) [inline bazel/envoy_build_system.bzl]: Please add a link to the gperftools bug report (here and in `envoy_test_linkopts`).

**Comment by jmillikin-stripe** [comment]
Looks good to me other than one comment.

**Comment by zuercher** [comment]
For reference the actual change is that previously bazel passed `-undefined dynamic_lookup` to the linker and with the new CROSSTOOL.tpl it omits the option altogether.

**Pull request merged**